app/scripts/controllers/provider-approval.js

```diff
@@ -48,7 +49,7 @@ class ProviderApprovalController extends SafeEventEmitter {
       if (approved) {
         res.result = [this.preferencesController.getSelectedAddress()]
       } else {
-        throw new Error('User denied account authorization')
+        throw rpcErrors.eth.userRejectedRequest('User denied account authorization.')
```

Comment:
Looks like we added a period here. If we're looking for maximum backwards-compat, we shouldn't.

Reply:
> I noticed one typo which is arguably breaking.

Resolved.

> I don't fully understand how `eth-json-rpc-errors` improves error messages, or how this fixes a former break, so it might be nice to also document those two things, even just here in the PR before approve/merge.

WIP.